### PR TITLE
Populate styling page

### PR DESCRIPTION
**Pull request by itskitto**
Added in the content from the Styling portion of the docs; changed the ordering of the content a bit to fit  what I think is a logical "getting started" flow.

Will need to get opinions on that change and would preferably get @drwpow's approval as well.

EDIT
Adding this here so we can reference it later
image

### CODE REVIEW

File: src/pages/basic-concepts/styling.md [hunk 1]
```diff
@@ -1,4 +1,510 @@
...
+<h1>I have both scoped and global styles</h1>
+```
+
+**Tips**
+
+- `<style>` tags within `.astro` files will be extracted and optimized for you on build. So you can write CSS without worrying too much about delivery.
+- For best result, only have one `<style>` tag per-Astro component. This isn’t necessarily a limitation, but it may result in better optimization at buildtime.
+- If you want to import third-party libraries into an Astro component, you can use [Sass][sass]! In particular, [@use][sass-use] may come in handy (e.g. `@use "bootstrap/scss/bootstrap"`);
+
+## Cross-Browser Compatibility
```
Comment (drwpow) [inline src/pages/basic-concepts/styling.md:40]: 👍🏻 I like moving this section here

File: src/pages/basic-concepts/styling.md [hunk 2]
```diff
@@ -1,4 +1,510 @@
...
+
+We also automatically add browser prefixes using [Autoprefixer][autoprefixer]. By default, Astro loads the [Browserslist defaults][browserslist-defaults], but you may also specify your own by placing a [Browserslist][browserslist] file in your project root.
+
+---
+
+## Supported Styling Options
```
Comment (drwpow) [inline src/pages/basic-concepts/styling.md:46]: Yeah this does flow better—addressing the core styling first, then frontend frameworks later.

**Review by drwpow — APPROVED** [review]
Looks good! Thanks for taking the time to improve this page 🙂

File: src/pages/basic-concepts/styling.md [hunk 3]
```diff
@@ -1,4 +1,510 @@
...
+
+```html
+<head>
+  <link rel="stylesheet" href="/styles/global.css" />
+</head>
+```
+
+And in your local filesystem, you can even use Sass’ [@use][sass-use] to combine files together effortlessly:
+
+```
```
Comment (jasikpark) [inline src/pages/basic-concepts/styling.md:269]: Would it make sense to add an example of utility CSS in use somewhere around here? Like an iteration on the above scoped CSS <button> that needs the addition of the utility CSS, or an example that uses multiple `<Button/>`s in a list and shows why `margin` for example is good as a utility class?

**Review by jasikpark — APPROVED** [review]
This is amazing! It's definitely a lot to read through, but the info in here is great!

File: src/pages/basic-concepts/styling.md [hunk 4]
```diff
@@ -1,4 +1,510 @@
...
+    /* ❌ This will fight with <Button>’s styles */
+  }
+</style>
+
+<nav class="nav">
+  <Button>Menu</Button>
+</nav>
+```
+
+This is undesirable because now `<Nav>` and `<Button>` fight over what the final button looks like. Now, whenever you edit one, you’ll always have to edit the other, and they are no longer truly isolated as they once were (now coupled by a bidirectional styling dependency). It’s easy to see how this pattern only has to repeated a couple times before being afraid that touching any styles _anywhere_ may break styling in a completely different part of the app (queue `peter-griffin-css-blinds.gif`).
```
Comment (jasikpark) [inline src/pages/basic-concepts/styling.md:218]: maybe add a knowyourmeme link?

File: src/pages/basic-concepts/styling.md (hunk 3 above)
Comment (drwpow) [reply src/pages/basic-concepts/styling.md:269]: That’s a good idea. While in general I think the docs could explain utility CSS better, I also think there’s so much to talk about there it deserves a larger section. E.g. would `<button class="mt1">` mean anything to people? Would they expect Bootstrap utility CSS? Bulma? Tailwind? (I even encourage writing your own utility classes at one point).

But if we had a larger explainer, then I’d love to reference back to that in more places like this

Comment (itskitto) [reply src/pages/basic-concepts/styling.md:269]: Agreed, I think as we get more of the documentation filled out with existing information, we all can formally meet about how things can be improved and broken up into more digestible pieces of information. A phase 2 sort of deal.

File: src/pages/basic-concepts/styling.md (hunk 4 above)
Comment (itskitto) [reply src/pages/basic-concepts/styling.md:218]: I'll gladly add it if you lead me to the right link 😄

Comment (jasikpark) [reply src/pages/basic-concepts/styling.md:218]: I couldn't find one D: this is the best I've got: https://css-tricks.com/can-we-create-a-resize-hack-with-container-queries/#h-can-we-debunk-a-classic-css-meme-with-container-queries

**Pull request merged**